File: Java/Leccion09/ContextoEstatico/Dominio/Persona.java
package ContextoEstatico.Dominio;

public class Persona {

    //Atributos
    private int idPersona;
    private static int contadorPersonas;
    private String nombre;

    //Constructor
    public Persona(String nombre) {
        this.nombre = nombre;
        //Incrementar el contador para cada objeto nuevo
        Persona.contadorPersonas++; //Acceso a variable estatica - Sintaxis correcta
        // Vamos asignar un nuevo valor a la variable idPersona
        this.idPersona = Persona.contadorPersonas; 
    }

    //Metodos getters y setters
    public static int getContadorPersonas() {
        return contadorPersonas;
    }

    public static void setContadorPersonas(int aContadorPersonas) {
        contadorPersonas = aContadorPersonas;
    }

    public int getIdPersona() {
        return this.idPersona;
    }

    public void setIdPersona(int idPersona) {
        this.idPersona = idPersona;
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    @Override //Metodo toString para imprimir los valores de los atributos, @Override significa sobreescribir, es decir, desde la herencia desde la clase Object se hereda éste método toString() y lo estamos sobreescribiendo, que recibimos desde la clase Object
    public String toString() {
        return "Persona{" + "idPersona=" + idPersona + ", nombre=" + nombre + '}';
    }

}

File: Python/Leccion10/Parte2/Persona.py
class Persona:

    contador_personas = 0 #Variable de clase
    @classmethod #Metodo de clase
    def generar_siguiente_valor(cls): 
        cls.contador_personas += 1
        return cls.contador_personas


    def __init__(self, nombre, edad):
        self.id_persona = Persona.generar_siguiente_valor()
        self.nombre = nombre
        self.edad = edad

    def __str__(self):
        return f'Persona [{self.id_persona} {self.nombre} {self.edad}]'
    
p1 = Persona('Juan', 28)
print(p1)
p2 = Persona('Karla', 25)
print(p2)
p3 = Persona('Carlos', 30)
print(p3)
Persona.generar_siguiente_valor() #Con el metodo de clase podemos incrementar el contador sin crear un objeto
p4 = Persona('María', 22)
print(p4)
print(f'Valor contador personas: {Persona.contador_personas}')
